agent/api/task.go

```diff
@@ -899,6 +900,24 @@ func (task *Task) updateContainerDesiredStatus() {
 	}
 }
 
+// updateResourceDesiredStatus sets all resources' desired status's depending on the
+// task's desired status
+func (task *Task) updateResourceDesiredStatus() {
+	for _, r := range task.Resources {
+		taskDesiredStatus := task.GetDesiredStatus()
+		if taskDesiredStatus == TaskRunning {
```

Comment:
Can you have a map that maps the task status to the resource status, so you don't rely on the actual status here?

Reply:
To use `TaskStatus` in `taskresource` package(for the mapping), we need to import `api` package. This is causing import cycle since `api` already imports `taskresource`. The plan was to move the status types from `api` into its own package. This is requiring changes in a lot of packages since `TaskStatus` and `ContainerStatus` and their types are used everywhere. I have added a `TODO` in this method, which shall be addressed in a different PR. Does that sound good?